src/scriptworker/context.py

```diff
@@ -231,12 +242,14 @@ async def populate_projects(self, force=False):
                 is already defined. Defaults to False.
 
         """
+        assert self.config
         if force or not self.projects:
             with tempfile.TemporaryDirectory() as tmpdirname:
                 self.projects = await load_json_or_yaml_from_url(self, self.config["project_configuration_url"], os.path.join(tmpdirname, "projects.yml"))
 
     @property
-    def download_semaphore(self):
+    def download_semaphore(self) -> asyncio.BoundedSemaphore:
+        assert self.config
```

Comment:
This causes the following test bustage:

```
________________________________ test_download_semaphore ________________________________

    @pytest.mark.asyncio
    async def test_download_semaphore():
        context = swcontext.Context()
>       sem = context.download_semaphore

tests/test_context.py:157:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <scriptworker.context.Context object at 0x110227460>

    @property
    def download_semaphore(self) -> asyncio.BoundedSemaphore:
>       assert self.config
E       AssertionError

src/scriptworker/context.py:252: AssertionError
```

Fairly easy fix of setting `context.config` to a non-empty dict in that test.